framework/src/DriverFramework.cpp

```diff
@@ -274,6 +267,9 @@ int Framework::initialize()
 		return -4;
 	}
 
+	// make sure the time offset is initialized
+	(void) offsetTime();
```

Comment:
I wasn't sure if it wasn't called (indirectly) earlier during initialization, so this is safer. But you're right, if we move it to the top, we can do it that way: https://github.com/PX4/DriverFramework/commit/5c693e3f9d0f314e407ad09b75b0a90833bc350d